plugins/modules/vmware_vc_infraprofile_info.py

```diff
@@ -0,0 +1,292 @@
+            json.dump(parsed, outfile, ensure_ascii=False, indent=2)
+        self.module.exit_json(changed=False, export_config_json=config_json)
+
+    def read_profile(self):
+        with open(self.config_path, "r") as file:
+            return file.read()
+
+    def get_import_profile_spec(self):
+        infra = self.api_client.appliance.infraprofile.Configs
+        config_spec = self.read_profile()
+        profile_spec = self.get_profile_spec()
+        import_profile_spec = infra.ImportProfileSpec(config_spec=config_spec, profile_spec=profile_spec)
+        return import_profile_spec
+
+    def vc_import_profile_task(self):
+        infra = self.api_client.appliance.infraprofile.Configs
+        import_profile_spec = self.get_import_profile_spec()
+        import_task = infra.import_profile_task(import_profile_spec)
+        self.wait_for_task(import_task)
+        if "SUCCEEDED" == import_task.get_info().status:
+            self.module.exit_json(changed=True, status=import_task.get_info().result.value)
+        self.module.fail_json(msg='Failed to import profile status:"%s" ' % import_task.get_info().status)
+
+    def vc_validate_profile_task(self):
+        infra = self.api_client.appliance.infraprofile.Configs
+        import_profile_spec = self.get_import_profile_spec()
+        validate_task = infra.validate_task(import_profile_spec)
+        if "VALID" == validate_task.get_info().result.get_field("status").value:
+            self.module.exit_json(changed=False, status=validate_task.get_info().result.get_field("status").value)
+        elif "INVALID" == validate_task.get_info().result.get_field("status").value:
+            # TO-DO: move to vmware_rest_client
+            self.module.exit_json(changed=False, status=validate_task.get_info().result.get_field("status").value)
+        else:
+            # TO-DO: move to vmware_rest_client
+            self.module.fail_json(msg='Failed to validate profile status:"%s" ' % dir(validate_task.get_info().status))
+
+    def wait_for_task(self, task, poll_interval=1):
+        while task.get_info().status == "RUNNING":
+            time.sleep(poll_interval)
+
+
+def main():
+    argument_spec = VmwareRestClient.vmware_client_argument_spec()
+    argument_spec.update(
+        encryption_key=dict(type='str', required=False),
+        description=dict(type='str', required=False),
+        decryption_key=dict(type='str', required=False),
+        api=dict(type='str', required=False, choices=['list', 'export', 'import', 'validate']),
+        profiles=dict(type='str', required=False),
+        config_path=dict(type='str', required=False),
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+    result = {'failed': False, 'changed': False}
+    vmware_vc_infra_profile = VcenterProfile(module)
+    vmware_vc_version = VcVersionChecker(module)
+    vmware_vc_version.check_vc_version()
+
+    if module.params['api'] == "list":
```

Comment:
use `lower()` function, it is better to have case insensitive choices
Similar for the below choices